### ISSUE TITLE
Journals needing scripts

### ISSUE THREAD

**Issue by willpearse** [ropensci/suppdata#2]
- Elsevier
- PNAS (...doesn't seem to have SIs other than PDFs, but check)
- Nature
- Frontiers in ...
- Cell (TREE et al.)
- ESA journals - not every one of them seems to use Archives

**Comment by willpearse** [comment]
`url <- paste0("http://www.pnas.org",.grep.url(paste0("http://www.pnas.org/lookup/doi/",doi), "(/content/)[0-9/]*"),".abstract")`
could help with PNAS

**Comment by willpearse** [comment]
Elsevier is officially unintelligible to me. I've no idea how the redirects are working, and I just can't do it :-(

**Comment by willpearse** [comment]
- [ ] Elsevier
- [ ] PNAS
- [ ] Nature
- [ ] Frontiers in XXX
- [ ] Cell (TREE etc.)
- [ ] Cambridge dissertations and other misc. (see #10 )

**Comment by rossmounce** [comment]
...as well as, in no particular order...

- [ ] **Pensoft**
28 journals all open access and on the same platform tech (ARPHA). Here's an example article from _Deutsche Entomologische Zeitschrift_ with two suppdata files: https://doi.org/10.3897/dez.65.21000  
a) https://doi.org/10.3897/dez.95.21000.suppl1 b) https://doi.org/10.3897/dez.95.21000.suppl2 

- [ ] **Ubiquity Press** 
?? journals, all (?) open access and on the same platform tech. Here's an example article from _Open Quaternary_  with one suppdata file: http://doi.org/10.5334/oq.32 a) https://doi.org/10.5334/oq.32.s1 

- [x] **Copernicus Publications**
 40 journals, all open access and on the same platform tech. Here's an example article from _Biogeosciences_ with one suppdata file: https://doi.org/10.5194/bg-14-1739-2017 a) https://www.biogeosciences.net/14/1739/2017/bg-14-1739-2017-supplement.zip 

- [ ] **Hindawi**
Over 250 open access journals all on the same platform tech. Here's an example article from _International Journal of Ecology_ with one suppdata file: https://doi.org/10.1155/2017/5620125 b) https://www.hindawi.com/journals/ijecol/2017/5620125/#supplementary-material-1 ▶️  http://downloads.hindawi.com/journals/ijecol/2017/5620125.f1.docx 

- [x] **MDPI**
Over 190 open access journals on the same platform tech. Here's an example article from _Molecules_ with one suppdata file: https://doi.org/10.3390/molecules22050681 a) http://www.mdpi.com/1420-3049/22/5/681#supplementary ▶️ http://www.mdpi.com/1420-3049/22/5/681/s1 

- [ ] **Cambridge University Press**
Over 380 journals on the same platform tech. Here's an example article from _Oryx_ with one suppdata file: https://doi.org/10.1017/S0030605315001246 a) https://www.cambridge.org/core/journals/oryx/article/taking-the-elephant-out-of-the-room-and-into-the-corridor-can-urban-corridors-work/4C43E703569F63FEA771EB61D327F4F2#supplementary-materials ▶️ `https://static.cambridge.org/resource/id/urn:cambridge.org:id:binary:20170303101053109-0421:S0030605315001246:S0030605315001246sup001.pdf` 

- [ ] **Taylor and Francis Online**
Loads of journals on the same platform. Here's an example article from _Zoology and Ecology_ with one suppdata file: https://doi.org/10.1080/21658005.2017.1366293 a) https://www.tandfonline.com/doi/suppl/10.1080/21658005.2017.1366293?scroll=top ▶️ https://doi.org/10.6084/m9.figshare.5350366 

- [ ] **BioOne**
Over 200 journals all on the same platform. Here's an example article from _Journal of Parasitology_ with one suppdata file: https://doi.org/10.1645/17-15 a) http://www.bioone.org/doi/suppl/10.1645/17-15 ▶️ http://www.bioone.org/doi/suppl/10.1645/17-15/suppl_file/10.1645_17-15.s1.docx 

- [ ] **SAGE Journals**
~1,000 journals all on the same platform. Here's an example article from _American Sociological Review_ with two suppdata files: https://doi.org/10.1177/0003122415621900 a) http://journals.sagepub.com/doi/suppl/10.1177/0003122415621900 ▶️ http://journals.sagepub.com/doi/suppl/10.1177/0003122415621900/suppl_file/england_online_supplement.pdf and http://journals.sagepub.com/doi/suppl/10.1177/0003122415621900/suppl_file/England_february_2016_media_abstracts.pdf 

- [ ] **Royal Society of Chemistry Publishing**
Over 80 journals on the same platform. Here's an example article from _Chemical Communications_ with one suppdata file:  https://doi.org/10.1039/C8CC01172A a) http://www.rsc.org/suppdata/c8/cc/c8cc01172a/c8cc01172a1.pdf

- [ ] **Company of Biologists (the not open access journals & articles)**
This publisher is somewhat complicated. It has five journals. The two open access ones _Disease Models & Mechanisms_ and _Biology Open_ can have their supplementary data accessed via the existing EuropePMC method. But for the three closed access journals _Development_ , _Journal of Cell Science_ and _Journal of Experimental Biology_ a specific per-publisher script is needed. Here's an example article from _Development_ with one suppdata file: https://doi.org/10.1242/dev.160382 a) http://dev.biologists.org/lookup/doi/10.1242/dev.160382.supplemental ▶️ http://dev.biologists.org/highwire/filestream/1248938/field_highwire_adjunct_files/0/DEV160382supp.pdf 
~~**BMJ Journals, eLife, F1000Research, Wellcome Open Research, Gates Open Research**~~ All covered via EuropePMC method

**Comment by rossmounce** [comment]
PS I would retitle the thread (is that possible on GH?) to 'publisher platforms needing scripts' as the problem is per platform tech, not per journal (thankfully!)

**Comment by rossmounce** [comment]
I see you've listed Cell Press as distinct from Elsevier? Why? 

You can access the Cell Press imprint journals through the ScienceDirect platform, like other non-Cell Press Elsevier journals. Why implement two distinct methods when one would suffice? Suggest you merge the two as "Elsevier inc. Cell Press journals (ScienceDirect)" or some such notation.

**Comment by nuest** [comment]
May I add some to the list?

- [x] [PeerJ](https://peerj.com/)
- [x] [JStatSoft](https://www.jstatsoft.org) (potentially with a generic function for OJS journals)

Are there any guidelines or step-by-step instructions for how to add a journal?

**Update: Working on _Copernicus Publications_ now!**

**Comment by willpearse** [comment]
PeerJ would be a great one to add, thanks!

The instructions are here: https://github.com/ropensci/suppdata/blob/master/CONTRIBUTING.md. If you need any help with it, please just ask (maybe make a separate issue or something). Thanks very much!

**Comment by nuest** [comment]
@willpearse can you check "Copernicus Publications" in the list in this post, please: https://github.com/ropensci/suppdata/issues/2#issuecomment-371476111

**Comment by nuest** [comment]
I think we need a nice list of already supported journals as well. I can't find one - am I just missing it? Happy to add one to the README file if I know it doesn't exist yet :-)

**Comment by willpearse** [comment]
We don't have one right now - if you could add it somewhere sensible in the README I'd be very grateful indeed. Thank you!

**Comment by nuest** [comment]
List added with #33

Just wondering how to prioritize here... Maybe we should create bounties for publishers, and send an e-mail to each of them?  :-) Or we as developers just vouch to add 1 source / month? That's one every quarter for three people @willpearse @sckott :-)

**Comment by sckott** [comment]
I imagine publishers won't help, and probably don't have any R people. I can put it on my list, but likely will be slower than 1 per month sadly

**Comment by willpearse** [comment]
I don't think publishers will help either, sadly. I think the one per month
thing is a good idea, Daniel: I think the best way to get more developers
to help is to show that we're doing it ourselves. I'll do PenSoft by
November 25th (unless you'd like it?).

>

**Comment by nuest** [comment]
No, go ahead. Just poke me when you're done, so I can pick up the next one.

@sckott My suggestion was only about once every three months :-)
